internal/controller/pgupgrade/pgupgrade_controller.go

```diff
@@ -93,7 +95,10 @@ func (r *PGUpgradeReconciler) findUpgradesForPostgresCluster(
 // Reconcile does the work to move the current state of the world toward the
 // desired state described in a [v1beta1.PGUpgrade] identified by req.
 func (r *PGUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, err error) {
-	log := ctrl.LoggerFrom(ctx)
+	ctx, span := tracing.Start(ctx, "reconcile-pgupgrade")
+	log := logging.FromContext(ctx)
+	defer span.End()
+	defer func(s tracing.Span) { _ = tracing.Escape(s, err) }(span)
```

Comment:
:thinking: This is another place where I am unsure why we're using `Escape` instead of `Check` (there are others, but I think it's generally the same question, so I won't mention any more).